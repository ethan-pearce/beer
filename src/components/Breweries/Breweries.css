.breweryName {
    text-align: left;
    font-size: 60px;
    font-weight: 300;
    background-color: #d8541d;
    width: fit-content;
    color: white;
    padding: 0 8px 8px;
    display: block;
    line-height: 50px;
}

.breweryName:hover {
    opacity: 0.8;
}

.breweryLink {
    text-align: left;
    margin: 8px 0 8px 8px;
    font-size: 32px;
    display: block;
    width: fit-content;
}


.breweryLink:hover {
    text-decoration: underline;
}

.breweryImage {
    display: block;
    text-align: left;
    height: 150px;
}