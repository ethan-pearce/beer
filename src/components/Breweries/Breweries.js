import './Breweries.css';
import { Link } from 'react-router-dom';
import Locations from '../Locations/Locations';

export const Breweries = ({breweries}) => {
    if(!breweries || breweries.length <= 0){
        return null;
    }

    return (
        <ul className={'breweries'}>
          {breweries.map(brewery => {
            const { href, name, image, locations, id } = brewery;
            return (
              <li key={id} className={'brewery'}>
                <Link to={`/brewery/${id}`} className={'breweryName'}>{name}</Link>
                <a className={'breweryLink'} href={href} rel="noreferrer" target="_blank">{href}</a>
                <img className={'breweryImage'} alt={name} src={image}/>
                <Locations locations={locations} breweryName={name}/>
              </li>
            )
          })}
      </ul>
    )
}

export default Breweries;