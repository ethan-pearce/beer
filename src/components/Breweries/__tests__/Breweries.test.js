import React from 'react'
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from "react-router-dom";
import Breweries from '../Breweries'

describe('Breweries Component', () => {
    const breweries = [
        {
            name: 'test name 1',
            id: '1',
            href: 'https://test.com/',
            image: 'https://test.com/test.jpg',
            locations: [
                {
                    name: 'test name 1',
                    id: '1',
                    type: 'test type 1',
                    region: 'test region 1',
                    postalCode: '00001'
                }
            ]
        }
    ];

    it('should render', () => {
        const { container } = render(<Router><Breweries breweries={breweries}/></Router>)

        expect(container.getElementsByClassName('breweries').length).toBe(1)
        expect(container.getElementsByClassName('brewery').length).toBe(1)
        expect(container.getElementsByClassName('LocationTitle').length).toBe(1)
        expect(container.getElementsByClassName('location').length).toBe(1)
        expect(container).toMatchSnapshot()
    })

    it('should render with no locations if none are passed', () => {
        breweries[0].locations = [];
        const { container } = render(<Router><Breweries breweries={breweries}/></Router>)

        expect(container.getElementsByClassName('breweries').length).toBe(1)
        expect(container.getElementsByClassName('brewery').length).toBe(1)
        expect(container.getElementsByClassName('LocationTitle').length).toBe(0)
        expect(container.getElementsByClassName('location').length).toBe(0)
        expect(container).toMatchSnapshot()
    })

    it('should return null if not breweries passed', () => {
        const { container } = render(<Router><Breweries /></Router>)

        expect(container.getElementsByClassName('breweries').length).toBe(0)
        expect(container.getElementsByClassName('brewery').length).toBe(0)
        expect(container).toMatchSnapshot()
    })
})