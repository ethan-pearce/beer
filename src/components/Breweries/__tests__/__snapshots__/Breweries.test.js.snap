// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breweries Component should render 1`] = `
<div>
  <ul
    class="breweries"
  >
    <li
      class="brewery"
    >
      <a
        class="breweryName"
        href="/brewery/1"
      >
        test name 1
      </a>
      <a
        class="breweryLink"
        href="https://test.com/"
        rel="noreferrer"
        target="_blank"
      >
        https://test.com/
      </a>
      <img
        alt="test name 1"
        class="breweryImage"
        src="https://test.com/test.jpg"
      />
      <p
        class="LocationTitle"
      >
        Locations:
      </p>
      <ul
        class="locations"
      >
        <li
          class="location"
        >
          <a
            href="https://www.google.com/maps/search/?api=1&query=test name 1+test name 1+test region 1+00001"
            rel="noreferrer"
            target="_blank"
          >
            <p>
              test name 1
            </p>
            <p>
              Type: test type 1
            </p>
            <p>
              test region 1
            </p>
            <p>
              00001
            </p>
          </a>
        </li>
      </ul>
    </li>
  </ul>
</div>
`;

exports[`Breweries Component should render with no locations if none are passed 1`] = `
<div>
  <ul
    class="breweries"
  >
    <li
      class="brewery"
    >
      <a
        class="breweryName"
        href="/brewery/1"
      >
        test name 1
      </a>
      <a
        class="breweryLink"
        href="https://test.com/"
        rel="noreferrer"
        target="_blank"
      >
        https://test.com/
      </a>
      <img
        alt="test name 1"
        class="breweryImage"
        src="https://test.com/test.jpg"
      />
    </li>
  </ul>
</div>
`;

exports[`Breweries Component should return null if not breweries passed 1`] = `<div />`;
